src/SDKs/ServiceBus/data-plane/tests/Microsoft.Azure.ServiceBus.Tests/Diagnostics/QueueClientDiagnosticsTests.cs

```diff
@@ -12,443 +12,463 @@ namespace Microsoft.Azure.ServiceBus.UnitTests.Diagnostics
 
     public sealed class QueueClientDiagnosticsTests : DiagnosticsTests
     {
-        protected override string EntityName => TestConstants.NonPartitionedQueueName;
-        private QueueClient queueClient;
+        private QueueClient queueClient;        
         private bool disposed = false;
 
         [Fact]
         [LiveTest]
         [DisplayTestMethodName]
         async Task EventsAreNotFiredWhenDiagnosticsIsDisabled()
         {
-            this.queueClient = new QueueClient(TestUtility.NamespaceConnectionString, TestConstants.NonPartitionedQueueName, ReceiveMode.ReceiveAndDelete);
-
-            Activity processActivity = null;
-
-            ManualResetEvent processingDone = new ManualResetEvent(false);
-            this.listener.Disable();
-
-            await TestUtility.SendMessagesAsync(this.queueClient.InnerSender, 1);
-            this.queueClient.RegisterMessageHandler((msg, ct) => {
-                    processActivity = Activity.Current;
-                    processingDone.Set();
-                    return Task.CompletedTask;
-                },
-                exArgs => Task.CompletedTask);
-
-            processingDone.WaitOne(TimeSpan.FromSeconds(maxWaitSec));
-            Assert.True(this.events.IsEmpty);
-            Assert.Null(processActivity);
+            await ServiceBusScope.UsingQueueAsync(partitioned: false, sessionEnabled: false, async queueName =>
```

Comment:
I know it doesn't work in all the cases but a would simplify a lot of them.

Reply:
I'm not a fan either, honestly.   That said, I looked at changing it and decided against doing so to avoid refactoring the `Dispose` approach for closing and cleaning things up.    Given the wide scope of changes, I'm inclined to just leave this as-is.